pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java

```diff
@@ -415,6 +633,34 @@ private void compactAssignmentTopic() {
         }
     }
 
+    protected int updateWorkerDrainMap() {
```

Comment:
Since this gets run periodically, lets make sure method cannot somehow get invoked concurrently.

Reply:
@kaushik-develop did you address this?